plugin/ipam/driver.go

```diff
@@ -44,10 +61,33 @@ func (i *ipam) ReleasePool(poolID string) error {
 	return nil
 }
 
-func (i *ipam) RequestAddress(poolID string, address net.IP, options map[string]string) (*net.IPNet, map[string]string, error) {
+func (i *ipam) RequestAddress(poolID string, address net.IP, options map[string]string) (ip *net.IPNet, _ map[string]string, err error) {
 	Log.Debugln("RequestAddress", poolID, address, options)
-	// Pass magic string to weave IPAM, which then stores the address under its own string
-	ip, err := i.weave.AllocateIP("_")
+	// If we pass magic string "_" to weave IPAM it stores the address under its own string
+	if poolID == "weavepool" { // old-style
+		ip, err = i.weave.AllocateIP("_")
+	} else if strings.HasPrefix(poolID, "weave-") {
+		dashPos := strings.Index(poolID[6:], "-")
+		if dashPos == -1 {
+			return nil, nil, fmt.Errorf("Unrecognized pool ID: %s", poolID)
+		}
+		subnetStr := poolID[6 : dashPos+6]
+		iprangeStr := poolID[dashPos+7:]
```

Comment:
using `strings.Split(...)` would make all the above a bit less fiddly, allowing you to get rid of all the indexing and even the `HasPrefix`, i.e. you'd simply split `poolID` on "-", check you get three elements back and that the first is "weave".